rose/engine.py

```diff
@@ -0,0 +1,487 @@
+# flake8: noqa
+import queue
+import time
+import threading
+
+from functools import wraps
+from typing import Callable, Dict
+from concurrent.futures import Future
+from concurrent.futures import ThreadPoolExecutor
+
+import radical.utils as ru
+import radical.pilot as rp
+
+import typeguard
+from .data import InputFile, OutputFile
+
+class Task(rp.TaskDescription):
+    """
+    Represents a task description by extending the `TaskDescription` class from `rp`
+    (an external module).
+
+    This class is primarily used to define and manage the details of a task, inheriting
+    properties and methods from the `rp.TaskDescription` base class. Additional arguments
+    and keyword arguments can be passed to further configure the task, which are then
+    forwarded to the base class.
+
+    Parameters
+    ----------
+    *args : tuple
+        Positional arguments to pass to the parent class constructor, if needed.
+    **kwargs : dict
+        Keyword arguments to configure the task. Passed directly to the `TaskDescription`
+        initializer.
+
+    Methods
+    -------
+    None. This class relies on inherited methods from `rp.TaskDescription`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(kwargs)
```

Comment:
Task class `__init__` does not allow args. As a user-friendly approach and for ROSE to do the following:

```python
Task(executable='foo.py')
```

instead of 
```python
td = Task()
td.executable = 'foo.py'
```

I pass the kwargs as:

```python
super().__init__(from_dict=kwargs)
```

And I will let Task._verify handle the verification process.

Reply:
> Task(executable='foo.py')

We could allow the `rp.TaskDescription` the same actually.  The underlying base class (`ru.TypedDict`) supports this.  Well, that is not for here and now though.

> Task class __init__ does not allow args

Well, that's just the point: then your `__init__` above should not allow `*args` either I think, because these would get dropped anyway.

Reply:
Already addressed here: https://github.com/radical-cybertools/ROSE/blob/1f14a41a70bade3996c2a77bc871146ebb5ff5b9/rose/engine.py#L40